cyclonedx/model/__init__.py

```diff
@@ -767,6 +770,28 @@ def deserialize(cls, o: Any) -> 'XsUri':
                 f'XsUri string supplied does not parse: {o!r}'
             ) from err
 
+    @classmethod
+    def make_bom_link(
+        cls,
+        serial_number: Union[UUID, str],
+        version: int = 1,
+        bom_ref: Optional[Union[str, BomRef]] = None
+    ) -> 'XsUri':
+        """
+        Generate a BOM-Link URI.
+
+        Args:
+            serial_number: The unique serial number of the BOM.
+            version: The version of the BOM. The default version is 1.
+            bom_ref: The unique identifier of the component, service, or vulnerability within the BOM.
+
+        Returns:
+            XsUri: Instance of XsUri with the generated BOM-Link URI.
+        """
+        bom_ref_part = f'#{quote(str(bom_ref))}' if bom_ref else ''
+        uri = f'urn:cdx:{serial_number}/{version}{bom_ref_part}'
+        return cls(uri)
+
```

Comment:
please add a method `is_bom_link(self) -> bool`.
every `XsUri` that statrts with `urn:cdx:` is a BOM-Link

Reply:
Do we want something trivial as this
```python3
    def is_bom_link(self) -> bool:
        return self._uri.startswith('urn:cdx:')
 ```
 
 or something complex as this
 ```python3
     _BOM_LINK_REGEX = re.compile(r'^urn:cdx:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/[1-9][0-9]*(#.+)?$')

    def is_bom_link(self) -> bool:
        return bool(self._BOM_LINK_REGEX.match(self._uri))
 ```

Reply:
the simple `.startswith('urn:cdx:')` looks sufficient for now.

Reply:
Done